src/services/ItemService/item.ts

```diff
@@ -14,3 +16,71 @@ export const createItemService = async (newItem: ItemType) => {
       throw new Error(error);
     });
 };
+
+
+
+export const getAllItems =async (req: Request, res: Response) => {
+  
+  const items = await new ItemRepository().AllItem();
+  
+  return items
+}
+
+
+export const getItemById = async (req:Request, res:Response) => {
+  
+  const item = await new ItemRepository().SingleItem(req.params.id)
+  if (item) {
+    return item
+  } else {
+    res.status(404)
+    throw new BadRequestError('Item not found')
+  }
+}
+
+export const updateItem = async (req:any, res:Response) => {
+  
+  const {
+    name,
+    price,
+    description,
+    image,
+    location,
+    category,
+    status,
+    shippingOptions,
+  } = req.body
+console.log(req.body);
+
+  const item = await new ItemRepository().editItem (req.params.id)
+
+  if (item) {
+    item.name = name
```

Comment:
What if the name and other details the user is parsing during update is empty, did you cater for that in your middleware??

Reply:
Added...